packages/react/src/headless.tsx

```diff
@@ -58,11 +59,28 @@ function Root(props: RootProps) {
 
   return (
     <MarkpromptContext.Provider value={contextValue}>
-      <Dialog.Root modal>{props.children}</Dialog.Root>
+      <DialogRootWithAbort>{props.children}</DialogRootWithAbort>
     </MarkpromptContext.Provider>
   );
 }
 
+function DialogRootWithAbort({ children }: { children: React.ReactNode }) {
+  const { abort } = useContext(MarkpromptContext);
+
+  return (
+    <Dialog.Root
+      modal
+      onOpenChange={(open) => {
+        if (!open) {
+          abort();
+        }
+      }}
```

Comment:
We should probably also call the user-provided `onOpenChange`, if any, so users can also perform actions on that state change.